src/scripts/add-user-to-group.ts

```diff
@@ -0,0 +1,121 @@
+        userMissing
+          ? `User [${user}] does not exist. Can not add to an existing group.`
+          : ''
+      }${userMissing && groupMissing ? '\n' : ''}${
+        groupMissing
+          ? `Group [${group}] does not exist. Can not add a user to a non-existent group.`
+          : ''
+      }`
+    );
+  };
+
+  // eslint-disable-next-line functional/functional-parameters
+  const throwOnAlreadyPresent = () => {
+    // eslint-disable-next-line functional/no-throw-statement
+    throw new Error('User already exists in group.');
+  };
+
+  return maybeOktaUser === undefined || maybeOktaGroup === undefined
+    ? throwOnMissing(maybeOktaUser === undefined, maybeOktaGroup === undefined)
+    : (await userExistsInGroup(maybeOktaGroup, maybeOktaUser.id))
+    ? throwOnAlreadyPresent()
+    : maybeOktaUser.addToGroup(maybeOktaGroup.id);
+};
+
+export default (
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  rootCommand: RootCommand
+): Argv<{
+  readonly clientId: string;
+  readonly privateKey: string;
+  readonly organisationUrl: string;
+  readonly user: string;
+  readonly group: string;
+}> =>
+  rootCommand.command(
+    'add-user-to-group [user] [group]',
+    'Adds an existing user to an existing group.',
+    // eslint-disable-next-line functional/no-return-void, @typescript-eslint/prefer-readonly-parameter-types
+    (yargs) => {
+      // eslint-disable-next-line functional/no-expression-statement
+      yargs
+        .option('user-id', {
+          type: 'string',
+          alias: ['user-login', 'user-email', 'user'],
+          // eslint-disable-next-line quotes
+          describe: "The user's ID, login, or email address",
+          demandOption: true,
+        })
+        .option('group', {
+          type: 'string',
+          // eslint-disable-next-line quotes
+          describe: "The group's ID",
+          demandOption: true,
+        });
+    },
+    async (args: {
+      readonly clientId: string;
+      readonly privateKey: string;
+      readonly organisationUrl: string;
+      readonly user: string;
+      readonly group: string;
+      // eslint-disable-next-line @typescript-eslint/require-await
+    }) => {
+      // eslint-disable-next-line functional/no-try-statement
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const response: Response = await addUserToGroup(
+          {
+            ...args,
+          },
+          args.user,
+          args.group
+        );
+        // eslint-disable-next-line functional/no-expression-statement
+        console.info(
+          // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions, @typescript-eslint/no-unsafe-member-access
+          response.ok
+            ? `Added user [${args.user}] to group [${args.group}].`
+            : `User [${args.user}] was not added to group [${args.group}] correctly, however no errors were caught when attempting to do so.`
```

Comment:
Why would this happen and what would an end user do if they saw this message?

Reply:
I don't know why it would happen, but there could be a potential for it to happen. Perhaps a bad internet connection?

Reply:
Better to have an error be thrown with whatever explanation response contains as to the cause then.

Reply:
Error will be thrown on a bad response.